keep/posthog/posthog.py

```diff
@@ -1,16 +1,47 @@
 import os
-
+import time
+import uuid
 import posthog
+import logging
+import requests
+
 from posthog import Posthog
 
-if os.getenv("DISABLE_POSTHOG", "false") == "true":
-    posthog.disabled = True
+logger = logging.getLogger(__name__)
 
+DISABLE_POSTHOG = os.getenv("DISABLE_POSTHOG", "false") == "true"
+UPTIME_REPORTING_CADENCE = 60 * 10
+RANDOM_TENANT_ID_PERSISTENT_WITHIN_LAUNCH = uuid.uuid4()
 
-def get_posthog_client():
+if DISABLE_POSTHOG:
+    posthog.disabled = True
+
+def get_posthog_client(sync_mode=False):
     posthog_api_key = (
         os.getenv("POSTHOG_API_KEY")
         or "phc_muk9qE3TfZsX3SZ9XxX52kCGJBclrjhkP9JxAQcm1PZ"
     )
-    posthog_client = Posthog(api_key=posthog_api_key, host="https://app.posthog.com")
+    posthog_client = Posthog(api_key=posthog_api_key, host="https://app.posthog.com", sync_mode=sync_mode)
```

Comment:
This will add delay to every request received by Keep

Reply:
if I understand it correctly

Reply:
It's only for `is_posthog_reachable` to raise Exception if posthog is not reachable. False by default.